src/lib/common/topology_sampler.cpp

```diff
@@ -53,6 +52,7 @@ typedef struct network_configuration_info {
 
 typedef struct node_list {
   uint64_t nodes[TOPOLOGY_SAMPLER_MAX_NODE_LIST_SIZE];
+  abstractSensorType_e sensor_type[TOPOLOGY_SAMPLER_MAX_NODE_LIST_SIZE];
```

Comment:
I created another list that will match the nodes (node_id) list such that element 0 in both lists will reference the same node.